src/libraries/react-calendar/providers/calendar-context/CalendarContext.tsx

```diff
@@ -0,0 +1,57 @@
+import {
+  ReactNode,
+  createContext,
+  useCallback,
+  useContext,
+  useState,
+} from 'react'
+import type { CalendarState, SetCalendarState } from './CalendarContext.type'
+
+const initialCalendarState: CalendarState = {
+  selectedDate: new Date(),
+  layout: 'week',
+  openDialog: false,
+  openViewDialog: false,
+  defaultEventDuration: 60,
+  modal: false,
+  eventDialogMaxWidth: 'md',
+  fullscreen: false,
+  allowFullScreen: false,
+  withCloseIcon: true,
+  title: '',
+  content: '',
+  actions: '',
+  draggingEventId: -1,
+  startDragging: false,
+  ghostProperties: { width: 0, height: 0, date: new Date() },
+}
+
+const CalendarStateContext = createContext<CalendarState>({} as CalendarState)
+
+const SetCalendarStateContext = createContext<SetCalendarState>(
+  {} as SetCalendarState
+)
+
+export const useCalendarState = () => useContext(CalendarStateContext)
+
+export const useSetCalendarState = () => useContext(SetCalendarStateContext)
```

Comment:
Those two hooks make a little sense, as they basically do nothing, general way of using context is to write `useContext` inside component itself. This gives you understating of what part of the application you are working with. In this case `useSetCalendarState` can reffer to completely different functionallity, for example changing state in local storage or in database or in some other service.